src/Item/Item.router.spec.ts

```diff
@@ -529,6 +533,13 @@ describe('Item router', () => {
               tpCollectionMock.urn_suffix,
               itemUrnSuffix
             )
+            const originalItemMock = { ...new Item() }
+            mockItem.mockImplementation((createdItem) => {
+              return {
+                ...originalItemMock,
+                upsert: jest.fn().mockResolvedValueOnce(createdItem),
+              } as any
+            })
```

Comment:
is it necessary to do this? I mean couldn't you just do `Item.update.mockImplementation`? It's reseted afterEach

Reply:
We need to test two things:
- The DB updates or inserts the item with the correct information.
- We're returning the upserted item.
To check that we're upserting the item correctly, the only way to do it is by mocking the constructor.

There's another alternative though, and that's to mock the return value of the `upsert` method and then do an assertion on the constructor of the item.